internal/alloydb/refresh.go

```diff
@@ -236,10 +220,10 @@ type refreshResult struct {
 	expiry         time.Time
 }
 
-type certChain struct {
-	root         *x509.Certificate
-	intermediate *x509.Certificate
-	client       *x509.Certificate
+type certs struct {
+	certChain *tls.Certificate // TLS client certificate
```

Comment:
We can just make this `tls.Certificate` and treat it as a value object, as all the internals are references anyway.

Reply:
Sounds good.

Reply:
Done